docs/Deprecation.md

```diff
@@ -4,6 +4,11 @@ When deprecating a function, class, type, or data constructor in the cardano-nod
 
 1. Use the [`DEPRECATED`](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/pragmas.html#warning-and-deprecated-pragmas) pragma.
 
+1. Add a comment stating deprecation date, for example:
+```haskell
+-- DEPRECATED SINCE 2023-05-01
+```
```

Comment:
Is date better?  Or version?  Also is it better into the `DEPRECATED` pragma message or comment?

Reply:
I think date could be potentially confusing because our releases don't necessarily come from `master` and we could have release from the `master` branch where the function isn't deprecated as of that date.  I think it is sufficient to mark something as deprecated.